src/Mappers/InventoryMapper.php

```diff
@@ -306,20 +347,21 @@ static function getSupplierPartNumberFromVariation($variationData, $itemMappingM
 
     $supplierPartNumber = null;
 
-    $mainVariationId = $variationData['id'];
-    $variationNumber = $variationData['number'];
+    $variationNumber = $variationData[self::VARIATION_COL_NUMBER];
 
     $supplierPartNumber = null;
 
     switch ($itemMappingMethod) {
       case AbstractConfigHelper::ITEM_MAPPING_SKU:
-        if (array_key_exists(self::VARIATION_SKUS, $variationData) && !empty($variationData[self::VARIATION_SKUS])) {
-          $supplierPartNumber = $variationData[self::VARIATION_SKUS][0]['sku'];
+        if (array_key_exists(self::VARIATION_COL_SKUS, $variationData) && !empty($variationData[self::VARIATION_COL_SKUS])) {
+          // FIXME: should not blindly use the first SKU - only should use the one assigned to Wayfair!
```

Comment:
This FIXME tag is related to the known issue reported here and not something new, correct? https://github.com/wayfair-contribs/plentymarkets-plugin/issues/154

Reply:
right. wanted to call it out in the code so that we keep an eye on it.